File: multimodal-generation/image-generation/python/02_color_guided_generation.py
#!/usr/bin/env python3

from random import randint
from amazon_image_gen import BedrockImageGenerator
import file_utils
import logging
from datetime import datetime
import base64

logging.basicConfig(level=logging.INFO, format="[%(levelname)s] %(message)s")
logger = logging.getLogger(__name__)


def main():
    # Load the reference image and encode it as a Base64 string.
    reference_image_path = "../images/color-guided-ref-image-1.png"
    with open(reference_image_path, "rb") as image_file:
        reference_image_base64 = base64.b64encode(image_file.read()).decode("utf-8")

    # Configure the inference parameters.
    inference_params = {
        "taskType": "COLOR_GUIDED_GENERATION",
        "colorGuidedGenerationParams": {
            "text": "digital painting of a girl, dreamy and ethereal",
            "colors": ["#81FC81", "#386739", "#C9D688", "#FFFFFF", "#FFFFFF"],
            "referenceImage": reference_image_base64,
        },
        "imageGenerationConfig": {
            "numberOfImages": 1,  # Number of variations to generate. 1 to 5.
            "quality": "standard",  # Allowed values are "standard" and "premium"
            "width": 1280,  # See README for supported output resolutions
            "height": 720,  # See README for supported output resolutions
            "cfgScale": 8.0,  # How closely the prompt will be followed
            "seed": randint(0, 858993459),  # Use a random seed
        },
    }

    # Define an output directory with a unique name.
    generation_id = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
    output_directory = f"output/{generation_id}"

    # Create the generator.
    generator = BedrockImageGenerator(output_directory=output_directory)

    # Generate the image(s).
    response = generator.generate_images(inference_params)

    if "images" in response:
        # Save each image
        file_utils.save_base64_images(response["images"], output_directory, "image")


if __name__ == "__main__":
    main()

File: multimodal-generation/image-generation/javascript/02_color_guided_generation.js
import { BedrockImageGenerator } from './BedrockImageGenerator.js';

const generateImages = async () => {
    // Format timestamp for unique directory naming
    const timestamp = new Date().toISOString().replace(/[:.]/g, '-').slice(0, 19);
    const outputDirectory = `output/${timestamp}`;

    const generator = new BedrockImageGenerator({ outputDirectory });

    const params = {
        taskType: 'COLOR_GUIDED_GENERATION',
        colorGuidedGenerationParams: {
            text: 'digital painting of a girl, dreamy and ethereal',
            colors: ['#81FC81', '#386739', '#C9D688', '#FFFFFF', '#FFFFFF'],
        },
        imageGenerationConfig: {
            numberOfImages: 1, // Number of variations to generate. 1 to 5.      
            quality: 'standard', // Allowed values are "standard" and "premium"
            width: 1280, // See README for supported output resolutions
            height: 720, // See README for supported output resolutions
            cfgScale: 7.0, // How closely the prompt will be followed
            seed: Math.floor(Math.random() * 858993459), // Use a random seed
        }
    };

    try {
        const images = await generator.generateImages(params);
        console.log('Generated images:', images.join(', '));
    } catch (err) {
        console.error('Image generation failed:', err.message);
        process.exit(1);
    }
};

// Self-executing async function
(async () => {
    try {
        await generateImages();
    } catch (err) {
        console.error('Fatal error:', err.message);
        process.exit(1);
    }
})();